### DISCUSSION TITLE
AADSTS90102: 'redirect_uri' value must be a valid absolute URI.

### DISCUSSION THREAD

**Discussion by henningwriedt** [abraunegg/onedrive#1743] [Q&A] [answered]
Anybody else who is encountering this issue when first trying to authorize the onedrive client?

"Message: AADSTS90102: 'redirect_uri' value must be a valid absolute URI."

Versions:
onedrive v2.4.14-6-ga2d2ea9
Linux elementary6 5.11.0-41-generic #45~20.04.1-Ubuntu

**Comment by abraunegg** [comment]
@henningwriedt 
This must be something unique to your environment & setup and/or account type.

What is the full URL that the application is asking you to access to authenticate?

**Comment by henningwriedt** [comment]
@abraunegg 

Interesting. I just recompiled a second ago according to the instructions on [Github](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md)

and still get the same error from the microsoft login web page. The URL I'm trying is: 
```
https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=d50ca740-c83f-4d1b-b616-12c519384f0c&scope=Files.ReadWrite%20Files.ReadWrite.all%20Sites.Read.All%20Sites.ReadWrite.All%20offline_access&response_type=code&prompt=login&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient
```

**Comment by abraunegg** [comment]
@henningwriedt 
OK - the URL looks OK / no issue. 100% suspect that this is related to your account type | account configuration.

What sort of account type is this:
* OneDrive Personal
  * Are you using an @outlook.com account or something else?
* OneDrive Business
  * Are you using a corporate email address that is has OneDrive enabled?
* Do you use / have you enabled any 2FA?
* Have you configured Windows Hello?

**Comment by henningwriedt** [comment]
1. outlook.com
2. actually there might be a deactivated business account - need to double-check - any advice on what to do?
3. 2FA - I sometimes used the app „Authenticator“ - do I need to disable this somewhere?
4. Windows Hello - never used that!

Thanx for your support!!

Beste Gruesse
Henning

**Comment by abraunegg** [comment]
@henningwriedt 
My best suggestion is as follows:
* Use Google Chrome to perform the authentication | login to authenticate the client - do not use any other type of browser under Linux - this could be the cause of the problem
* You are using an OS based on Ubuntu - my advice there is try on something that is *not* Ubuntu or Ubuntu based ... 
* Raise a case with Microsoft. It is their API | authentication process that is failing you - nothing to do with this client at that stage of client operation if the above 2 points dont resolve your issue

**Comment by henningwriedt** [comment]
1. I‘m using Chrome.
2. Will try with a manjaro or debian tomorrow.

Thanks again!

Beste Gruesse
Henning Wriedt

**Comment by henningwriedt** [comment, marked as answer]
Thanks @abraunegg !

Just to follow up so others might benefit as well: I installed the client on Manjaro 5.10.83 via "pamac build onedrive-abraunegg" from the Arch Linux User Repository (AUR) like in the install guide mentioned. No problem to authenticate whatsoever using Chromium.

This led me to try something different: Use a browser (in this case firefox oddly enough) that I never used to authenticate against MS Onedrive - and it worked! So I assume it is a good advice to clear the browser (password) cache and retry.

(Happy to be able to use your great software again)
Henning

**Comment by abraunegg** [comment]
@henningwriedt 
Glad this was something simple in the end in regards to browser cache - although the actual error message from Microsoft is totally misleading!
